lib/SILGen/SILGenFunction.h

```diff
@@ -590,23 +593,29 @@ class LLVM_LIBRARY_VISIBILITY SILGenFunction
   StringRef getMagicFunctionString();
 
   /// Enter the debug scope for \p Loc, creating it if necessary.
```

Comment:
Perhaps add something like:

`/// \arg isGuardScope marks a scope introduced by a guard let statement. Guard let may used to unwrap an optional and bind it to the same name (as in guard let x = x!). All guard let scopes are popped when their parent lexical scope is popped from DebugScopeStack.`